src/dataflow-types/src/types.rs

```diff
@@ -51,13 +50,30 @@ impl PeekResponse {
     pub fn unwrap_rows(self) -> Vec<(Row, NonZeroUsize)> {
         match self {
             PeekResponse::Rows(rows) => rows,
-            PeekResponse::Error(_) | PeekResponse::Canceled => {
+            PeekResponse::Error(_) => {
                 panic!("PeekResponse::unwrap_rows called on {:?}", self)
             }
         }
     }
 }
 
+/// Peek responses that the controller can provide back to the coordinator.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum CoordPeekResponse {
```

Comment:
This is a bit of an abstraction violation—can this type get hoisted into the `coord` crate?

Reply:
I've also realized it's just...wrong. The controller never sends a Cancel, so it should still be sending a regular PeekResponse, and only the coord gets to see a CoordPeekResponse. Will move.